modules/gitops-architecture-argocd-agent-comparison.adoc

```diff
@@ -0,0 +1,48 @@
+// Module included in the following assemblies:
+//
+// * argocd-agent-architecture-overview/gitops-architecture-argocd-agent-comparison.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="gitops-architecture-argocd-agent-comparison_{context}"]
+= Comparing Argo CD Agent with traditional Argo CD architecture
+
+The Argo CD Agent architecture offers several advantages and tradeoffs compared to the traditional push-based hub-and-spoke model.
+
+[cols="2a,2a,2a",options="header"]
+|===
+| Capability
+| Traditional push-based Argo CD
+| Argo CD Agent pull-based
+
+| Single pane of glass
+| Does not provide a `single pane of glass` to monitor Argo CD Application resources across multiple Argo CD instances.
+| Provides a single `single pane of glass` to monitor all Argo CD Application resources across all managed Argo CD instances.
+
+| Network connectivity
+| The `hub` (Argo CD) must connect directly to the destination cluster Kubernetes APIs. Disconnected or firewalled clusters cannot be managed.
+| The `hub` (Argo CD) enables deployments to disconnected or firewalled clusters through pull-based synchronization.
+
+| Scalability
+| A single Argo CD instance manages multiple destination clusters, leading to CPU and memory bottlenecks.
+| Cluster management is not performed centrally by the `hub` (Argo CD); each child cluster is managed by its own dedicated Argo CD deployment.
+
+| Security
+| The `hub` (Argo CD) stores credentials for all workload clusters, increasing the attack surface.
```

Comment:
JGW-TODO: `Additionally, workload clusters must expose their Kubernetes API endpoints to the control plane, increasing the attack surface on that cluster.`